dask_grblas/expr.py

```diff
@@ -15,6 +15,141 @@ def __init__(self, parent, method_name, *args, meta, **kwargs):
         self.kwargs = kwargs
         self._meta = meta
 
+    def _matmult(self, meta, updating=None, mask=None, accum=None):
+        a = self.parent._delayed
+        b = self.args[0]._delayed
+        op = self.args[1]
+        lhs_ind = 'ij' if (a.ndim == 2) else 'j'
+        rhs_ind = 'jk' if (b.ndim == 2) else 'j'
+        if lhs_ind == 'ij':
+            out_ind = 'ik' if (rhs_ind == 'jk') else 'i'
+        else:
+            out_ind = 'k' if (rhs_ind == 'jk') else ''
+
+        not_updating = updating is None
+        no_mask = mask is None
+        grblas_mask_type = None
+        if not_updating:
+            args = [a, lhs_ind, b, rhs_ind]
+        elif no_mask:
+            args = [updating, out_ind,
+                    a, lhs_ind, b, rhs_ind]
+        else:
+            grblas_mask_type = get_grblas_type(mask)
+            args = [updating, out_ind,
+                    mask.mask._delayed, out_ind,
+                    a, lhs_ind, b, rhs_ind]
+
+        return da.core.blockwise(
+            partial(_matmul, op, meta.dtype,
+                    not_updating,
+                    no_mask, grblas_mask_type,
+                    accum),
+            out_ind,
+            *args,
+            meta=wrap_inner(meta)
+        )
+
+    def _matmul2(self, meta, updating=None, mask=None, accum=None):
```

Comment:
This is a reimplementation of the matmul operation that mirrors the dask implementation.  To facilitate this implementation, a new chunk-type called FakeInnerTensor wraps InnerMatrix and is used to 'deceive' dask into thinking the chunk has 3 or more dimensions.

Reply:
I discovered later on (see [here](https://github.com/dask/dask/pull/8423#issue-1065134223)).  That `FakeInnerTensor` need only have 3 dimensions, no more.